backend/index.ts

```diff
@@ -1,5 +1,40 @@
-import type { _PublicKeyAuthenticatorProtocol } from '../types/interface'
-export interface CreateAuthenticatorOptions {}
-export function createPublicKeyAuthenticator(opts: CreateAuthenticatorOptions): _PublicKeyAuthenticatorProtocol {
-    return {}
+import type { PublicKeyAuthenticatorProtocol } from '../types/interface'
+
+export interface NormalizedCreateOptions {
+    keys: CryptoKeyPair
+    timeout: number
+    rpID: string
+    challenge: BufferSource
+    crossOrigin: boolean
+}
+
+export interface CreateAuthenticatorOptions {
+    getNormalizedCreateOptions(): Promise<NormalizedCreateOptions>
+
+    // sign count
+    getSignCount(key: CryptoKey): Promise<number>
+
+    incrementSignCount(key: CryptoKey): Promise<void>
+
+    hasCredential(options: PublicKeyCredentialCreationOptions | PublicKeyCredentialRequestOptions): Promise<boolean>
+
+    // without username
+    getResidentKeyPair(rpID: string): Promise<readonly [key: CryptoKeyPair, credentialID: ArrayBuffer]>
+
+    // with username
+    getKeyPairByKeyWrap(
```

Comment:
There should also has a method that's create the new crypto key and the return the credential ID by key wrap

Reply:
`credentialIDs` could be an empty array then it will create a new crypto key.

Reply:
Yes, actually... But that's a bit vague. I'd suggest a more explicit way